tools/bin/ecflow/processes/main/meteorological/get_med_metfcst.py

```diff
@@ -0,0 +1,194 @@
+    met_ds.tmmx.attrs['cell_methods'] = "time: maximum(interval: 24 hours)"
+    for var in ['tmmn', 'tmmx']:
+        met_ds[var].attrs['units'] = "degC"
+        met_ds[var].attrs['height'] = "2 m"
+    # wind speed
+    met_ds.wind_speed.attrs['units'] = "m/s"
+    met_ds.wind_speed.attrs['description'] = "Daily Mean Wind Speed"
+    met_ds.wind_speed.attrs['height'] = "10 m"
+    # shortwave radiation
+    met_ds.surface_downwelling_shortwave_flux_in_air.attrs['units'] = "W m-2"
+    met_ds.surface_downwelling_shortwave_flux_in_air.attrs['description'] = (
+        'Daily Mean Downward Shortwave Radiation At Surface')
+    # specific humidity
+    met_ds.specific_humidity.attrs['units'] = "kg/kg"
+    met_ds.specific_humidity.attrs['description'] = "Daily Mean Specific " + \
+        "Humidity"
+    met_ds.specific_humidity.attrs['height'] = "2 m"
+    return met_ds
+
+
+def main():
+    ''' Download meteorological forecast data for 90-day forecast
+        from http://thredds.northwestknowledge.net:8080/thredds/catalog/
+        NWCSC_INTEGRATED_SCENARIOS_ALL_CLIMATE/cfsv2_metdata_90day/catalog.html
+    '''
+    # read in configuration file
+    parser = argparse.ArgumentParser(description='Download met forecast data')
+    parser.add_argument('config_file', metavar='config_file',
+                        help='configuration file')
+    args = parser.parse_args()
+    config_dict = read_config(args.config_file)
+
+    # initialize cdo
+    cdo = Cdo()
+
+    # read in meteorological data location
+    met_fcst_loc = config_dict['MED_FCST']['Met_Loc']
+    old_config_file = config_dict['ECFLOW']['old_Config']
+    new_config_file = config_dict['ECFLOW']['new_Config']
+
+    # read in grid_file from config file
+    grid_file = config_dict['DOMAIN']['GridFile']
+
+    # define variable names used when filling threads URL
+    # an abbreviation and a full name is needed
+    varnames = {'vs': 'wind_speed', 'tmmx': 'air_temperature',
+                'tmmn': 'air_temperature',
+                'srad': 'surface_downwelling_shortwave_flux_in_air',
+                'sph': 'specific_humidity', 'pr': 'precipitation_amount',
+                'pet': 'potential_evapotranspiration'}
+    # define model names for file name
+    modelnames = ['CFSv2']
+    # ['NCAR', 'NASA', 'GFDL', 'GFDL-FLOR', 'ENSMEAN', 'CMC1', 'CMC2',
+    #              'CFSv2']
+    new_units = {'vs': 'm s-1', 'sph': 'kg kg-1',
+                 'tmmx': 'degC', 'tmmn': 'degC', 'srad': 'W m-2',
+                 'pr': 'mm', 'pet': 'mm'}
+    old_units = {'vs': 'm s-1', 'sph': 'kg kg-1',
+                 'tmmx': 'K', 'tmmn': 'K', 'srad': 'W m-2',
+                 'pr': 'mm', 'pet': 'mm'}
+
+    # download metdata from http://thredds.northwestknowledge.net
+    for model in modelnames:
+        dlist = []
+        for var, name in varnames.items():
+            url = ('http://thredds.northwestknowledge.net:8080/thredds/dodsC/' +
```

Comment:
any reason to not have this hardcoded?

Reply:
I can move this to the configuration file, but the attributes that are manually replaced and the variable names and units are specific to this data source. My opinion is that moving the url to a configuration file would imply that the script was available for general use when it is pretty specific.